### ISSUE TITLE
JSON.stringify is not the best way to compare objects

### ISSUE THREAD

**Issue by karlguillotte** [xiaolin/react-image-gallery#450]
Hi @xiaolin, 

Thanks for your great library!

I use your library to show images dragged & dropped by users. Using the `FileReader` API, I read the content of every file and set a property on my items that will eventually be consumed by the image gallery component. 

I recently upgraded to 0.9 and a change to the items comparaison introduces in 0.9.1 break my application. Using `JSON.stringify` breaks when it tries to stringify the output of the `FileReader.readAsDataURL` (JSON.stringify cannot serialize cyclic structures). 

Also, I would like to point out that using `JSON.stringify` to compare objects is proned to bugs: 

```js
JSON.stringify({ foo:1, bar:2 }) === JSON.stringify({ bar:2, foo:1 })  // false, even if they have the same values. 
```
In such case, using `Object.is` recursively or `lodash/isEqual` are probably more appropiate. 

https://github.com/xiaolin/react-image-gallery/blob/master/src/ImageGallery.jsx#L216

I reverted back to 0.8.x in the meantine. 

Thanks!

**Comment by xiaolin** [comment]
added commit https://github.com/xiaolin/react-image-gallery/commit/fe2105f945832197f2523bf8304618d4c1ac28dc.
needs release.

**Comment by xiaolin** [comment]
released in version >= v1.0.0
